packages/cms-cli/commands/init.js

```diff
@@ -20,18 +24,58 @@ const { addLoggerOptions, setLogLevel } = require('../lib/commonOpts');
 const { logDebugInfo } = require('../lib/debugInfo');
 
 const COMMAND_NAME = 'init';
+const HS_AUTH_OAUTH_COMMAND = 'hs auth oauth2';
 
 const promptUser = async promptConfig => {
   const prompt = inquirer.createPromptModule();
   return prompt(promptConfig);
 };
 
+const oauthConfigSetup = () => {
+  try {
+    createEmptyConfigFile();
+    const authProcess = spawn(HS_AUTH_OAUTH_COMMAND, {
+      stdio: 'inherit',
+      shell: true,
+    });
+
+    authProcess.on('close', deleteEmptyConfigFile);
+  } catch (e) {
+    logErrorInstance(e, HS_AUTH_OAUTH_COMMAND);
+  }
+
+  trackCommandUsage(COMMAND_NAME, {
+    authType: 'oauth',
+  });
+};
+
+const apiKeyConfigSetup = async ({ configPath }) => {
+  const configData = await promptUser([PORTAL_NAME, PORTAL_ID, PORTAL_API_KEY]);
+
+  try {
+    writeNewPortalApiKeyConfig(configData);
+  } catch (err) {
+    logFileSystemErrorInstance(err, {
+      filepath: configPath,
+      configData,
+    });
+  }
+
+  trackCommandUsage(COMMAND_NAME, {
+    authType: 'apiKey',
+  });
+};
+
 function initializeConfigCommand(program) {
   program
     .version(version)
     .description(
       `Initialize ${DEFAULT_HUBSPOT_CONFIG_YAML_FILE_NAME} for a HubSpot portal`
     )
+    .option(
+      '--api',
+      `create ${DEFAULT_HUBSPOT_CONFIG_YAML_FILE_NAME} using API key for authentication`
+    )
```

Comment:
Can we make the choice a part of the prompt flow?

Reply:
I added a list selection prompt for choosing the authentication method as well as the option to pass flags(--api/--oauth) to override the prompt.